Building from source fails to build hls-eval-plugin

<!--
If you encounter a bug or you have a support question, please try to fill out some of the information below.
Generally speaking, the information below is meant to help debugging issues but is no prerequisite for opening an issue.
-->
The package version requirements for `Diff` are too lax. Building requires `>=0.4.0`.

Which OS do you use:
Ubuntu 20.04

### Steps to reproduce
Clone `haskell-language-server` master. Run `./cabal-hls-install hls-8.10.4`

### Expected behaviour
Build should complete successfully.

### Actual behaviour
Build fails:
```sh
Failed to build hls-eval-plugin-1.0.0.0.
Build log (
/home/parallels/.cabal/logs/ghc-8.10.4/hls-eval-plugin-1.0.0.0-11dbf647e96e17fc298ace5eea655051522c471f829a0a52d82274c254ae5ffb.log
):
Configuring library for hls-eval-plugin-1.0.0.0..
Preprocessing library for hls-eval-plugin-1.0.0.0..
Building library for hls-eval-plugin-1.0.0.0..
[1 of 8] Compiling Ide.Plugin.Eval.Parse.Option ( src/Ide/Plugin/Eval/Parse/Option.hs, dist/build/Ide/Plugin/Eval/Parse/Option.o, dist/build/Ide/Plugin/Eval/Parse/Option.dyn_o )
[2 of 8] Compiling Ide.Plugin.Eval.Types ( src/Ide/Plugin/Eval/Types.hs, dist/build/Ide/Plugin/Eval/Types.o, dist/build/Ide/Plugin/Eval/Types.dyn_o )
[3 of 8] Compiling Ide.Plugin.Eval.Parse.Comments ( src/Ide/Plugin/Eval/Parse/Comments.hs, dist/build/Ide/Plugin/Eval/Parse/Comments.o, dist/build/Ide/Plugin/Eval/Parse/Comments.dyn_o )
[4 of 8] Compiling Ide.Plugin.Eval.Code ( src/Ide/Plugin/Eval/Code.hs, dist/build/Ide/Plugin/Eval/Code.o, dist/build/Ide/Plugin/Eval/Code.dyn_o )

src/Ide/Plugin/Eval/Code.hs:9:56: error:
    Module ‘Data.Algorithm.Diff’ does not export ‘PolyDiff(..)’
  |
9 | import           Data.Algorithm.Diff            (Diff, PolyDiff (..), getDiff)
  |                                                        ^^^^^^^^^^^^^
cabal: Failed to build hls-eval-plugin-1.0.0.0 (which is required by
exe:haskell-language-server from haskell-language-server-1.0.0.0). See the
build log above for details.

Error when running Shake build system:
  at want, called at src/Development/Shake/Internal/Args.hs:83:69 in shake-0.18.5-1e2d5c19198b250e49e59618f65b8507112b579e0eb0fa27f2e56c48ce5495a5:Development.Shake.Internal.Args
* Depends on: hls-8.10.4
  at command, called at src/Cabal.hs:29:13 in hls-install-0.8.0.0-inplace:Cabal
* Raised the exception:
Development.Shake.command, system command failed
Command line: cabal v2-install exe:haskell-language-server exe:haskell-language-server-wrapper -w /nix/store/lx0cx2xf38494pf85kmd6lc7wn8r16ry-ghc-8.10.4-with-packages/bin/ghc-8.10.4 --write-ghc-environment-files=never --installdir /home/parallels/.cabal/bin --max-backjumps=5000 --overwrite-policy=always --project-file=cabal.project -v1
Exit code: 1
Stderr:
Warning: Requested index-state2021-03-29T21:23:14Z is newer than
'hackage.haskell.org'! Falling back to older state (2021-03-29T13:57:59Z).
cabal: Failed to build hls-eval-plugin-1.0.0.0 (which is required by
exe:haskell-language-server from haskell-language-server-1.0.0.0). See the
build log above for details.
```

--- a/ghcide/ghcide.cabal
+++ b/ghcide/ghcide.cabal
@@ -84,7 +84,7 @@ library
         utf8-string,
         vector,
         hslogger,
-        Diff,
+        Diff ^>=0.4.0,
         vector,
         bytestring-encoding,
         opentelemetry >=0.6.1,

--- a/hls-plugin-api/hls-plugin-api.cabal
+++ b/hls-plugin-api/hls-plugin-api.cabal
@@ -4,7 +4,10 @@ version:       1.1.0.0
 synopsis:      Haskell Language Server API for plugin communication
 description:
   Please see the README on GitHub at <https://github.com/haskell/haskell-language-server#readme>
-homepage:      https://github.com/haskell/haskell-language-server/hls-plugin-api
+
+homepage:
+  https://github.com/haskell/haskell-language-server/hls-plugin-api
+
 bug-reports:   https://github.com/haskell/haskell-language-server/issues
 license:       Apache-2.0
 license-file:  LICENSE
@@ -32,38 +35,42 @@ library
     Ide.PluginUtils
     Ide.Types
 
-  hs-source-dirs:   src
+  hs-source-dirs:     src
   build-depends:
     , aeson
     , base                  >=4.12    && <5
     , containers
     , data-default
-    , Diff
-    , lsp                   ^>=1.2.0
+    , dependent-map
+    , dependent-sum
+    , Diff                  ^>=0.4.0
+    , dlist
     , hashable
     , hslogger
     , lens
+    , lsp                   ^>=1.2.0
+    , opentelemetry
     , process
     , regex-tdfa            >=1.3.1.0
     , shake                 >=0.17.5
     , text
     , unordered-containers
-    , dependent-map
-    , dependent-sum
-    , dlist
-    , opentelemetry
 
   if os(windows)
-      build-depends:
-        Win32
-  else
-      build-depends:
-        unix
+    build-depends: Win32
 
-  ghc-options:      -Wall -Wredundant-constraints -Wno-name-shadowing -Wno-unticked-promoted-constructors
+  else
+    build-depends: unix
+
+  ghc-options:
+    -Wall -Wredundant-constraints -Wno-name-shadowing
+    -Wno-unticked-promoted-constructors
 
   if flag(pedantic)
     ghc-options: -Werror
 
-  default-language: Haskell2010
-  default-extensions: DataKinds, KindSignatures, TypeOperators
+  default-language:   Haskell2010
+  default-extensions:
+    DataKinds
+    KindSignatures
+    TypeOperators

--- a/plugins/hls-eval-plugin/hls-eval-plugin.cabal
+++ b/plugins/hls-eval-plugin/hls-eval-plugin.cabal
@@ -19,11 +19,11 @@ build-type:         Simple
 extra-source-files:
   LICENSE
   README.md
-  test/testdata/*.yaml
-  test/testdata/*.hs
-  test/testdata/*.expected
-  test/testdata/*.lhs
   test/testdata/*.cabal
+  test/testdata/*.expected
+  test/testdata/*.hs
+  test/testdata/*.lhs
+  test/testdata/*.yaml
 
 flag pedantic
   description: Enable -Werror
@@ -53,7 +53,7 @@ library
     , base                  >=4.12    && <5
     , containers
     , deepseq
-    , Diff
+    , Diff                  ^>=0.4.0
     , directory
     , dlist
     , extra
@@ -111,7 +111,6 @@ test-suite tests
   build-tool-depends: hls-eval-plugin:test-server -any
   hs-source-dirs:     test
   main-is:            Main.hs
-
   build-depends:
     , aeson
     , base

--- a/plugins/hls-hlint-plugin/hls-hlint-plugin.cabal
+++ b/plugins/hls-hlint-plugin/hls-hlint-plugin.cabal
@@ -2,7 +2,9 @@ cabal-version: 2.2
 name:          hls-hlint-plugin
 version:       1.0.0.1
 synopsis:      Hlint integration plugin with Haskell Language Server
-description:   Please see Haskell Language Server Readme (https://github.com/haskell/haskell-language-server#readme)
+description:
+  Please see Haskell Language Server Readme (https://github.com/haskell/haskell-language-server#readme)
+
 license:       Apache-2.0
 license-file:  LICENSE
 author:        The Haskell IDE Team
@@ -23,29 +25,29 @@ flag ghc-lib
     Force dependency on ghc-lib-parser even if GHC API in the ghc package is supported
 
 library
-  exposed-modules:  Ide.Plugin.Hlint
-  hs-source-dirs:   src
+  exposed-modules:    Ide.Plugin.Hlint
+  hs-source-dirs:     src
   build-depends:
     , aeson
-    , apply-refact             >=0.9
-    , base                     >=4.12 && <5
+    , apply-refact          >=0.9
+    , base                  >=4.12    && <5
     , binary
     , bytestring
     , containers
     , data-default
     , deepseq
-    , Diff
+    , Diff                  ^>=0.4.0
     , directory
     , extra
     , filepath
-    , ghc-exactprint           >=0.6.3.4
-    , ghcide                   ^>= 1.1.0.0
+    , ghc-exactprint        >=0.6.3.4
+    , ghcide                ^>=1.1.0.0
     , hashable
-    , lsp
-    , hlint                    >=3.2
-    , hls-plugin-api           >= 1.0 && < 1.2
+    , hlint                 >=3.2
+    , hls-plugin-api        >=1.0     && <1.2
     , hslogger
     , lens
+    , lsp
     , regex-tdfa
     , shake
     , temporary
@@ -53,21 +55,25 @@ library
     , transformers
     , unordered-containers
 
-  if (!flag(ghc-lib) && impl(ghc >=8.10.1) && impl(ghc <9.0.0))
-    build-depends: ghc ^>= 8.10
+  if ((!flag(ghc-lib) && impl(ghc >=8.10.1)) && impl(ghc <9.0.0))
+    build-depends: ghc ^>=8.10
 
   else
     build-depends:
       , ghc
-      , ghc-lib            ^>= 8.10.4.20210206
-      , ghc-lib-parser-ex  ^>= 8.10
+      , ghc-lib            ^>=8.10.4.20210206
+      , ghc-lib-parser-ex  ^>=8.10
 
     cpp-options:   -DHLINT_ON_GHC_LIB
 
-  ghc-options:      -Wall -Wredundant-constraints -Wno-name-shadowing -Wno-unticked-promoted-constructors
+  ghc-options:
+    -Wall -Wredundant-constraints -Wno-name-shadowing
+    -Wno-unticked-promoted-constructors
 
   if flag(pedantic)
     ghc-options: -Werror
 
-  default-language: Haskell2010
-  default-extensions: DataKinds, TypeOperators
+  default-language:   Haskell2010
+  default-extensions:
+    DataKinds
+    TypeOperators

